### PR TITLE
Refactor cli code

### PR DESCRIPTION
**Pull request by alexandrebodin**
> ⚠️  includes code from https://github.com/strapi/strapi/pull/19175

### What does it do?

- refactor the strapi cli package org 
- flatten file structure when possible
- fix some naming issues

### Why is it needed?

slowly preparing for a new command api

### How to test it?

test you can run any strapi command as before in an example project

### CODE REVIEW

File: packages/core/strapi/src/cli/index.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import { createLogger } from './utils/logger';
+import { loadTsConfig } from './utils/tsconfig';
+import { CLIContext } from './types';
+
+const createCLI = async (argv: string[], command = new Command()) => {
+  try {
+    // NOTE: this is a hack to allow loading dts commands without make dts a dependency of strapi and thus avoiding circular dependencies
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const dtsCommands = require(require.resolve('@strapi/data-transfer')).commands;
```
Comment (Convly) [inline packages/core/strapi/src/cli/index.ts:13]: Do we still need this since data-transfer now depends on `core/core` & not `core/strapi`? (or is it?)

Comment (alexandrebodin) [reply packages/core/strapi/src/cli/index.ts:13]: Yes we still need to load the dts commands in the cli. but now data-transfer and strapi/strapi are not in a circle

**Review by joshuaellis — APPROVED** [review]
I've not tested, but the code changes look good. Does any of this need documenting in contributor docs i.e. adding new commands? Or is it idiomatic?

**Comment by alexandrebodin** [comment]
> I've not tested, but the code changes look good. Does any of this need documenting in contributor docs i.e. adding new commands? Or is it idiomatic?

I'll check I don't remember seening a doc at all about it

**Comment by alexandrebodin** [comment]
@joshuaellis found a few things to update :)

**Review by Convly — APPROVED** [review]
From what I've tested, it works as it should :)

**Pull request merged**